aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -532,7 +532,7 @@ Tensor frobenius_norm(const Tensor& self, IntList dim, bool keepdim) {
       dim.size(),
       " dimensions instead.");
   if (dim.size() == 1) {
-    return at::norm(self, 2, dim[0], keepdim);
+    return at::norm(self, 2, dim, keepdim, self.type().scalarType());
```

Comment:
Nyeh, let's please not do this. We've got `self.scalar_type()`.

Actually, I am wondering if `norm` should take ScalarType at all. `to()` takes `caffe2::TypeMeta` which is the more modern type (it's an open enum rather than a closed enum class.)

Reply:
All ScalarType in this file is accessed through type().scalarType() :rofl:. I'll update them all.

Having a ScalarType for norm allows us to fuse the conversion inside the kernel. I think it's always a win?

Reply:
Having ScalarType/TypeMeta is not a problem; we should have it. Just asking you to use TypeMeta rather than ScalarType ;)

And yes, we need to fix these :) But maybe in a different PR.

Reply:
Exposing TypeMeta in aten native functions seems to be really tricky.
jit is complaining, I'm now stuck with:
```volume/pytorch/torch/csrc/autograd/generated/python_torch_functions.cpp:5343:67: error: 'caffe2::typemeta' is not a class member
     return wrap(dispatch_norm(r.tensor(0), r.scalarOptional(1), r.caffe2::typemeta(2)));
```

I have put my temporary work here: https://github.com/jjsjann123/pytorch/commit/e78b434bb136f72b4ef7c038218a15848eece24b

Reply:
Got it compiled but jit is core dumping (Not surprising as I still haven't figured out many details on the code gen).
I think exposing TypeMeta in aten native functions is too involved to be squeezed into this PR. We can do this in a future PR is needed.